sdks/python/apache_beam/io/gcsio.py

```diff
@@ -547,6 +548,7 @@ def __init__(self, client, path, mime_type='application/octet-stream'):
     # Start uploading thread.
     self.upload_thread = threading.Thread(target=self._start_upload)
     self.upload_thread.daemon = True
+    self.upload_thread.last_error = None
     self.upload_thread.start()
 
   # TODO(silviuc): Refactor so that retry logic can be applied.
```

Comment:
Yes. TODO applies to the retrying whole of _start_upload(). Otherwise Insert() had already built-in retries, but it was considering the errors from _refresh to be not retriable. That is why we added a retry there. However, we still don't control the overall retry mechanics of the _start_upload().